### ISSUE TITLE
Flaky test Microsoft.CodeAnalysis.VisualBasic.UnitTests.GetExtendedSemanticInfoTests.GroupBy_Lookup1

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#23483]
The test `Microsoft.CodeAnalysis.VisualBasic.UnitTests.GetExtendedSemanticInfoTests.GroupBy_Lookup1` is failing frequently on merges from `master` to other branches. It isn't clear what the failure is (I tried testing the affected branch and did not see the failure).

The test is skipped to unblock the code flow. It needs to be investigated and unskipped.

Here is the output from the failing test:

``` none
Failed

Microsoft.CodeAnalysis.VisualBasic.UnitTests.GetExtendedSemanticInfoTests.GroupBy_Lookup1 (from Roslyn.Compilers.VisualBasic.Semantic.UnitTests)

MESSAGE:
  | Expected and actual were different.\r\nExpected:\r\nITranslatedQueryOperation (OperationKind.TranslatedQuery, Type: ?, IsInvalid) (Syntax: 'From s1 In ... in qi Group')\r\n Expression: \r\n IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n Children(4):\r\n IInvalidOperation (OperationKind.Invalid, Type: ?, IsImplicit) (Syntax: 's2 in qi')\r\n Children(3):\r\n ILocalReferenceOperation: qi (OperationKind.LocalReference, Type: QueryAble, IsInvalid) (Syntax: 'qi')\r\n IAnonymousFunctionOperation (Symbol: Function (s1 As System.Int32) As ?) (OperationKind.AnonymousFunction, Type: null, IsImplicit) (Syntax: 'qi')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsImplicit) (Syntax: 'qi')\r\n IReturnOperation (OperationKind.Return, Type: null, IsImplicit) (Syntax: 'qi')\r\n ReturnedValue: \r\n ILocalReferenceOperation: qi (OperationKind.LocalReference, Type: QueryAble) (Syntax: 'qi')\r\n IAnonymousFunctionOperation (Symbol: Function (s1 As System.Int32, s2 As System.Int32) As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n ReturnedValue: \r\n IAnonymousObjectCreationOperation (OperationKind.AnonymousObjectCreation, Type: &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>, IsImplicit) (Syntax: 's2 in qi')\r\n Initializers(2):\r\n IParameterReferenceOperation: s1 (OperationKind.ParameterReference, Type: System.Int32, IsImplicit) (Syntax: 's1')\r\n IParameterReferenceOperation: s2 (OperationKind.ParameterReference, Type: System.Int32, IsImplicit) (Syntax: 's2')\r\n IAnonymousFunctionOperation (Symbol: Function ($VB$It As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) As ?) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n ReturnedValue: \r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '')\r\n Children(0)\r\n IAnonymousFunctionOperation (Symbol: Function ($VB$It As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) As ?) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n ReturnedValue: \r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '')\r\n Children(0)\r\n IAnonymousFunctionOperation (Symbol: Function ($315 As ?, $VB$ItAnonymous As ?) As &lt;anonymous type: Key $315 As ?, Key $315 As ?>) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n ReturnedValue: \r\n IAnonymousObjectCreationOperation (OperationKind.AnonymousObjectCreation, Type: &lt;anonymous type: Key $315 As ?, Key $315 As ?>, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n Initializers(2):\r\n IOperation: (OperationKind.None, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n Children(1):\r\n IParameterReferenceOperation: $VB$ItAnonymous (OperationKind.ParameterReference, Type: ?, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\nActual:\r\nITranslatedQueryOperation (OperationKind.TranslatedQuery, Type: ?, IsInvalid) (Syntax: 'From s1 In ... in qi Group')\r\n Expression: \r\n IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n Children(4):\r\n IInvalidOperation (OperationKind.Invalid, Type: ?, IsImplicit) (Syntax: 's2 in qi')\r\n Children(3):\r\n ILocalReferenceOperation: qi (OperationKind.LocalReference, Type: QueryAble, IsInvalid) (Syntax: 'qi')\r\n IAnonymousFunctionOperation (Symbol: Function (s1 As System.Int32) As ?) (OperationKind.AnonymousFunction, Type: null, IsImplicit) (Syntax: 'qi')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsImplicit) (Syntax: 'qi')\r\n IReturnOperation (OperationKind.Return, Type: null, IsImplicit) (Syntax: 'qi')\r\n ReturnedValue: \r\n ILocalReferenceOperation: qi (OperationKind.LocalReference, Type: QueryAble) (Syntax: 'qi')\r\n IAnonymousFunctionOperation (Symbol: Function (s1 As System.Int32, s2 As System.Int32) As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'From s1 In qi, s2 in qi')\r\n ReturnedValue: \r\n IAnonymousObjectCreationOperation (OperationKind.AnonymousObjectCreation, Type: &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>, IsImplicit) (Syntax: 's2 in qi')\r\n Initializers(2):\r\n IParameterReferenceOperation: s1 (OperationKind.ParameterReference, Type: System.Int32, IsImplicit) (Syntax: 's1')\r\n IParameterReferenceOperation: s2 (OperationKind.ParameterReference, Type: System.Int32, IsImplicit) (Syntax: 's2')\r\n IAnonymousFunctionOperation (Symbol: Function ($VB$It As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) As ?) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n ReturnedValue: \r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '')\r\n Children(0)\r\n IAnonymousFunctionOperation (Symbol: Function ($VB$It As &lt;anonymous type: Key s1 As System.Int32, Key s2 As System.Int32>) As ?) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n ReturnedValue: \r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '')\r\n Children(0)\r\n IAnonymousFunctionOperation (Symbol: Function ($315 As ?, $VB$ItAnonymous As ?) As &lt;anonymous type: Key $315 As ?, Key $315 As ?>) (OperationKind.AnonymousFunction, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n IBlockOperation (1 statements) (OperationKind.Block, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n IReturnOperation (OperationKind.Return, Type: null, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n ReturnedValue: \r\n IAnonymousObjectCreationOperation (OperationKind.AnonymousObjectCreation, Type: &lt;anonymous type: Key $315 As ?, Key $315 As ?>, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n Initializers(2):\r\n IParameterReferenceOperation: $315 (OperationKind.ParameterReference, Type: ?, IsInvalid, IsImplicit) (Syntax: '')\r\n IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid, IsImplicit) (Syntax: '')\r\n Children(1):\r\n IParameterReferenceOperation: $VB$ItAnonymous (OperationKind.ParameterReference, Type: ?, IsInvalid, IsImplicit) (Syntax: 'Group ')\r\n
  | +++++++++++++++++++
  | STACK TRACE:
  | at Roslyn.Test.Utilities.AssertEx.Fail(String message) at Roslyn.Test.Utilities.AssertEx.AreEqual[T](T expected, T actual, String message, IEqualityComparer`1 comparer) at Microsoft.CodeAnalysis.Test.Utilities.OperationTreeVerifier.Verify(String expectedOperationTree, String actualOperationTree) at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.VerifyOperationTree(Compilation compilation, SyntaxNode node, String expectedOperationTree) at Microsoft.CodeAnalysis.VisualBasic.UnitTests.GetExtendedSemanticInfoTests.GroupBy_Lookup1()
```

**Comment by gafter** [comment]
@AlekseyTs Who would be the best person to investigate this? If the problem persists, this test may fail in the `features/compiler` branch after the current integration from `master` completes.

**Comment by gafter** [comment]
@heejaechang @mavasani Pulling the conversation here from https://github.com/dotnet/roslyn/pull/23484

**Comment by mavasani** [comment]
@AlekseyTs - I thought you added the IOperation verification to this test recently, and hence would be the best person to handle merge conflict resolution here.

**Comment by AlekseyTs** [comment]
Please update the baseline, I believe the diff
```
<  IOperation: (OperationKind.None, Type: null, IsInvalid, IsImplicit) (Syntax: '')
 > IParameterReferenceOperation: $315 (OperationKind.ParameterReference, Type: ?, IsInvalid, IsImplicit) (Syntax: '')
```

is expected doe to a fix merged into dev15.5.x https://github.com/dotnet/roslyn/pull/23224

**Comment by gafter** [comment]
@AlekseyTs If that fix is in master I would expect the test to have been failing in master. But it was passing in master. That suggests that the fix did not make it into master cleanly, perhaps due to some kind of merge conflict that git thought was OK but was not OK. Can you please investigate whether your fix #23224 made it into master or not?

**Comment by AlekseyTs** [comment]
The portion of the test was checked into master by https://github.com/dotnet/roslyn/pull/23343
A different fix was checked into dev15.5.x #23224, which is expected to affect the test once they meet.
It doesn't look like dev15.5.x was merged into master.

@gafter What I do not understand is why the test was disabled in master (where it was passing) rather than in a branch where it was failing. This doesn't make sense to me.

**Comment by AlekseyTs** [comment]
dev15.5.x was merged into master-vs-deps

**Comment by gafter** [comment]
> What I do not understand is why the test was disabled in master (where it was passing) rather than in a branch where it was failing. This doesn't make sense to me.

The test was not failing in any branch. The test was failing when we attempted to merge master into a number of other branches. The existence of the test in master was blocking integrations.

**Comment by AlekseyTs** [comment]
> The test was failing when we attempted to merge master into a number of other branches. 

If I was doing the integration, I would either figure out what is going on and update the base-line as part of the integration, or I would disable the test **as part of the integration**. That is the first time I see tests being disabled in source branch because integration breaks them in a destination branch. We should not be doing things like that in the future. Please enable the test in master because there is nothing wrong with it and we should not be jeopardizing quality of the product by disabling perfectly good tests due to problems elsewhere.
